dipy/viz/ui.py

```diff
@@ -2161,3 +2163,185 @@ def handle_move_callback(self, i_ren, obj, slider):
+        polys.InsertNextCell(4)
+        polys.InsertCellPoint(0)
+        polys.InsertCellPoint(1)
+        polys.InsertCellPoint(2)
+        polys.InsertCellPoint(3)
+        self.texture_polydata.SetPolys(polys)
+
+        tc = vtk.vtkFloatArray()
+        tc.SetNumberOfComponents(2)
+        tc.SetNumberOfTuples(4)
+        tc.InsertComponent(0, 0, 0.0)
+        tc.InsertComponent(0, 1, 0.0)
+        tc.InsertComponent(1, 0, 1.0)
+        tc.InsertComponent(1, 1, 0.0)
+        tc.InsertComponent(2, 0, 1.0)
+        tc.InsertComponent(2, 1, 1.0)
+        tc.InsertComponent(3, 0, 0.0)
+        tc.InsertComponent(3, 1, 1.0)
+        self.texture_polydata.GetPointData().SetTCoords(tc)
+
+        texture_mapper = vtk.vtkPolyDataMapper2D()
+        if major_version <= 5:
+            texture_mapper.SetInput(self.texture_polydata)
+        else:
+            texture_mapper.SetInputData(self.texture_polydata)
+
+        image = vtk.vtkTexturedActor2D()
+        image.SetMapper(texture_mapper)
+
+        self.texture = vtk.vtkTexture()
+        image.SetTexture(self.texture)
+
+        image_property = vtk.vtkProperty2D()
+        image_property.SetOpacity(1.0)
+        image.SetProperty(image_property)
+        self.actor = image
+
+        # Add default events listener to the VTK actor.
+        self.handle_events(self.actor)
+
+    def _get_actors(self):
+        """ Returns the actors that compose this UI component.
+        """
+        return [self.actor]
+
+    def _add_to_renderer(self, ren):
+        """ Add all subcomponents or VTK props that compose this UI component.
+
+        Parameters
+        ----------
+        ren : renderer
+        """
+        ren.add(self.actor)
+
+    def resize(self, size):
+        """ Resize the image.
+
+        Parameters
+        ----------
+        size : (float, float)
+            image size (width, height) in pixels.
+        """
+        # Update actor.
+        self.texture_points.SetPoint(0, 0, 0, 0.0)
+        self.texture_points.SetPoint(1, size[0], 0, 0.0)
+        self.texture_points.SetPoint(2, size[0], size[1], 0.0)
+        self.texture_points.SetPoint(3, 0, size[1], 0.0)
+        self.texture_polydata.SetPoints(self.texture_points)
+
+    def _set_position(self, coords):
+        """ Position the lower-left corner of this UI component.
+
+        Parameters
+        ----------
+        coords: (float, float)
+            Absolute pixel coordinates (x, y).
+        """
+        self.actor.SetPosition(*coords)
+
+    def scale(self, factor):
+        """ Scales the image.
+
+        Parameters
+        ----------
+        factor : (float, float)
+            Scaling factor (width, height) in pixels.
+        """
+        self.resize(self.size * factor)
+
+    def set_img(self, img):
+        """ Modifies the image used by the vtkTexturedActor2D.
+
+        Parameters
+        ----------
+        img : imageDataGeometryFilter
+
+        """
+        if major_version <= 5:
```

Comment:
Same as above, use something like `vtk_major_version` instead of `major_version`

Reply:
Should I change this for all classes?

Reply:
Sure, that would be great

Reply:
I'll do it in a separate PR using the `set_input` function from utils